pinot-spi/src/main/java/org/apache/pinot/spi/utils/TimestampUtils.java

```diff
@@ -60,8 +67,30 @@ public static long toMillisSinceEpoch(String timestampString) {
       try {
         return Long.parseLong(timestampString);
       } catch (Exception e1) {
-        throw new IllegalArgumentException(String.format("Invalid timestamp: '%s'", timestampString));
+        try {
+          return dateTimeToTimestamp(timestampString).getTime();
+        } catch (Exception e2) {
+          throw new IllegalArgumentException(String.format("Invalid timestamp: '%s'", timestampString));
+        }
+      }
+    }
+  }
+
+  /**
+   * Infers a date time format from a valid {@link Timestamp} string.
+   */
+  private static Timestamp dateTimeToTimestamp(String timestampString) {
```

Comment:
This will get called a lot on the critical path, so the point about `java.time` vs SDF is important.